src/snowflake/cli/api/utils/definition_rendering.py

```diff
@@ -370,17 +369,11 @@ def on_cycle_action(node: Node[TemplateVar]):
 
     project_definition = build_project_definition(**definition)
 
-    # Dump the model instead of using "definition" since model
-    # validators can change field values. Round-trip as JSON
-    # to serialize non-scalar types like Path to strings
-    project_context[CONTEXT_KEY] = json.loads(
-        project_definition.model_dump_json(
-            exclude_none=True,
-            warnings=False,
-            by_alias=True,
-            round_trip=True,
-        )
-    )
+    # Use the values originally provided by the user as the template context
+    # This intentionally doesn't reflect any field changes made by
+    # validators, to minimize user surprise when templating values
+    project_context[CONTEXT_KEY] = definition
```

Comment:
this undoes a previous change in this PR, restoring to the current behaviour from main